### ISSUE TITLE
Make clone_on_copy lint machine applicable

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust-clippy#4826]
Copy/pasted from https://github.com/rust-lang/rustfix/issues/182:

This gives a warning instead of replacing it with the suggested code:

```
$ cargo +nightly fix -Z unstable-options --clippy
warning: using `clone` on a `Copy` type
   --> src/parse/mod.rs:258:27
    |
258 |                 location: self.next_location().clone(),
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try dereferencing it: `*self.next_location()`
```

It would be really nice to have this be done automatically, since I currently have about 20 of these warnings in my code.

```sh
$ cargo +nightly --version
cargo 1.40.0-nightly (5da4b4d47 2019-10-28)
$ cargo +nightly clippy -V
clippy 0.0.212 (b4f17697 2019-11-14)
```

**Comment by jyn514** [comment]
cc @Manishearth

**Comment by flip1995** [comment]
In theory the suggestion is already fixable by rustfix:
https://github.com/rust-lang/rust-clippy/blob/86b8643586aa39f36fb7a02e98c8d64d31415e70/clippy_lints/src/methods/mod.rs#L1721-L1725

Only thing left to do here is to figure out the applicability level. (And maybe pass it through the `snippet_with_applicability` function)

**Comment by Manishearth** [comment]
Also write rustfix tests, we should avoid having any fixable lints without
tests
